packages/main/cypress/specs/SegmentedButton.cy.tsx

```diff
@@ -0,0 +1,239 @@
+			.should("not.have.attr", "selected");
+
+		cy.realPress("Enter");
+
+		cy.get<SegmentedButtonItem>("@secondItem")
+			.should("have.attr", "selected");
+	});
+
+	it("should select second item with space", () => {
+		cy.mount(
+			<>
+				<SegmentedButton>
+					<SegmentedButtonItem id="first">First</SegmentedButtonItem>
+					<SegmentedButtonItem id="second">Second</SegmentedButtonItem>
+				</SegmentedButton>
+			</>
+		);
+
+		cy.get("[ui5-segmented-button]")
+			.as("segmentedButton");
+
+		cy.get<SegmentedButton>("@segmentedButton")
+			.find("ui5-segmented-button-item")
+			.as("items");
+
+		cy.get<SegmentedButtonItem>("@items")
+			.first()
+			.realClick();
+
+		cy.get<SegmentedButtonItem>("@items")
+			.first()
+			.as("firstItem");
+
+		cy.get<SegmentedButtonItem>("@firstItem")
+			.should("be.focused");
+
+		cy.realPress("ArrowRight");
+
+		cy.get<SegmentedButtonItem>("@items")
+			.eq(1)
+			.as("secondItem");
+
+		cy.get<SegmentedButtonItem>("@secondItem")
+			.should("be.focused")
+			.should("not.have.attr", "selected");
+
+		cy.realPress("Space");
+
+		cy.get<SegmentedButtonItem>("@secondItem")
+			.should("have.attr", "selected");
+	});
+
+	it("should select last item with mouse", () => {
+		cy.mount(
+			<>
+				<SegmentedButton>
+					<SegmentedButtonItem id="first">First</SegmentedButtonItem>
+					<SegmentedButtonItem id="second">Second</SegmentedButtonItem>
+					<SegmentedButtonItem id="third">Third</SegmentedButtonItem>
+				</SegmentedButton>
+			</>
+		);
+
+		cy.get("[ui5-segmented-button]")
+			.as("segmentedButton");
+
+		cy.get<SegmentedButton>("@segmentedButton")
+			.find("ui5-segmented-button-item")
+			.as("items");
+
+		cy.get<SegmentedButtonItem>("@items")
+			.last()
+			.as("lastItem");
+
+		cy.get<SegmentedButtonItem>("@lastItem")
+			.ui5SegmentedButtonItemToggleSelect();
+	});
+
+	it("should be able to select multple items in multiple selection mode", () => {
+		cy.mount(
+			<>
+				<SegmentedButton selection-mode="Multiple">
+					<SegmentedButtonItem selected id="first">First</SegmentedButtonItem>
+					<SegmentedButtonItem id="second">Second</SegmentedButtonItem>
+				</SegmentedButton>
+			</>
+		);
+
+		cy.get("[ui5-segmented-button]")
+			.as("segmentedButton");
+
+		cy.get<SegmentedButton>("@segmentedButton")
+			.find("ui5-segmented-button-item")
+			.as("items");
+
+		cy.get<SegmentedButtonItem>("@items")
+			.first()
+			.as("firstItem");
+
+		cy.get<SegmentedButtonItem>("@items")
+			.last()
+			.as("lastItem");
+
+		// Select second item
+		cy.get<SegmentedButtonItem>("@lastItem")
+			.ui5SegmentedButtonItemToggleSelect();
+
+		// First item should still be selected
+		cy.get<SegmentedButtonItem>("@firstItem")
+			.should("have.attr", "selected");
+	});
+
+	it("should be able to deselect items in multiple selection mode", () => {
+		cy.mount(
+			<>
+				<SegmentedButton selection-mode="Multiple">
+					<SegmentedButtonItem selected id="first">First</SegmentedButtonItem>
```

Comment:
We should set explicitly boolean attributes `selected={true}`, please apply everywhere in the test